src/util/deleteThought.js

```diff
@@ -65,18 +69,19 @@ export const deleteThought = () => {
   // prev must be calculated before dispatching existingThoughtDelete
   const prev = showContexts
     ? prevContext()
-    : prevSibling(value, context, rank)
+    : prevSibling(state, value, context, rank)
 
   // returns true when thought is not hidden due to being a function or having a =hidden attribute
   const isVisible = thoughtRanked => state.showHiddenThoughts || (
     !isFunction(thoughtRanked.value) &&
-    !meta(context.concat(thoughtRanked.value)).hidden
+    !meta(state, context.concat(thoughtRanked.value)).hidden
   )
 
+  // must call store.getState() to use the new state after existingThoughtDelete
   const next = perma(() => showContexts
-    ? unroot(getContextsSortedAndRanked(headValue(contextOf(path))))[0]
+    ? unroot(getContextsSortedAndRanked(state, headValue(contextOf(path))))[0]
     // get first visible thought
-    : (sortPreference === 'Alphabetical' ? getThoughtsSorted : getThoughtsRanked)(context)
+    : (sortPreference === 'Alphabetical' ? getThoughtsSorted : getThoughtsRanked)(store.getState(), context)
```

Comment:
Is there any specific reason for using `store.getState()` here instead of `state`?